### DISCUSSION TITLE
MissingSchemaError | Nextjs + Mongoose

### DISCUSSION THREAD

**Discussion by Jossdz** [vercel/next.js#14598] [Help]
Hi!

I'm deploying my application on Vercel and works well the first time I connect, after a few minutes it throws this error:

![image](https://user-images.githubusercontent.com/23535629/85886038-ef940500-b7aa-11ea-9174-1bc4b7b31985.png)

This is how I'm connecting to db
![carbon (2)](https://user-images.githubusercontent.com/23535629/85886100-0b97a680-b7ab-11ea-997b-e4879226b1c2.png)

And here you have and example of my Models: 

![carbon (3)](https://user-images.githubusercontent.com/23535629/85886182-29650b80-b7ab-11ea-8055-aa7d22205208.png)

**Comment by hoangvvo** [comment]
This is my really, really **wild** guess and I may not be making much sense. The default export of `mongoose` package is an instance of `Mongoose` (See [here](https://github.com/Automattic/mongoose/blob/master/lib/index.js#L1117)). 

In serverful scenario, since every places import from the same package, they will all receive the same instance. You call `Mongoose.models` in one file and the `mongoose` in the other files will acknowledge that because they are all the same instance. This is not the case with serverless where every file/function is independent from each other so each might not have imported the same `Mongoose` instance.

Again, a really wild guess that may not make sense.

This is kinda the same thing that `Passport` does.
